persistence/persistence.go

```diff
@@ -17,6 +20,7 @@ type StateObject interface {
 // Base is the base for persistent objects.  All StateObjects should embed
 // Base and call NewBase to initialize.
 type Base struct {
+	// With datastore, this will show up in both the key and the Name field.
 	Name string
```

Comment:
This is a prefix string that we will have to parse. Please can we persist structured data?